src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -1037,7 +1037,10 @@ trait Types
      *  @param stableOnly     If set, return only members that are types or stable values
      */
     def findMember(name: Name, excludedFlags: Long, requiredFlags: Long, stableOnly: Boolean): Symbol = {
-      def findMemberInternal = new FindMember(this, name, excludedFlags, requiredFlags, stableOnly).apply()
+      def findMemberInternal = findMemberInstance.using { findMember =>
+        findMember.init(this, name, excludedFlags, requiredFlags, stableOnly)
```

Comment:
If I understand correctly, what this is doing is to have (mostly) a single instance of a single class, and we just re-set its fields. 

Wouldn't that make any method that uses those instances non-parallel and non-reentrant?

Reply:
It's not parallelizable, but neither is anything else in the compiler. The `ReusableInstance` object takes care of reentrancy.